### ISSUE TITLE
Fix FTA generation 

### ISSUE THREAD

**Issue by Matthew-Kulich** [kbss-cvut/fta-fmea-ui#101]
FaultEvent `func 111 fails` should not be `EventType.BASIC`. 
Look at this please and tell me if I should fix this or someone else. @kostobog 

<img width="599" alt="image" src="https://user-images.githubusercontent.com/67843655/196436790-398c60af-f36f-4562-8e29-d71f54cfdc2c.png">

**Comment by kostobog** [comment]
Yes the `func 111 fails` should not be a `EventType.BASIC`. Yes I agree this due to reuse of fault event from another tree. For now let's solve this issue by not reusing fault events across different fault trees.
